### PR TITLE
Fix potential memory leaks

### PR DESCRIPTION
**Pull request by AtariDreams**
These leaks relate to CFRelease and how in some edge cases, memory may leak.

This PR fixes that.

### CODE REVIEW

File: backend/usb-darwin.c [hunk 1]
```diff
@@ -1416,7 +1416,7 @@ static kern_return_t load_classdriver(CFStringRef	    driverPath,
   CFStringRef	bundle = driverPath ? driverPath : kUSBGenericTOPrinterClassDriver;
   char 		bundlestr[1024];	/* Bundle path */
   CFURLRef	url;			/* URL for driver */
-  CFPlugInRef	plugin = NULL;		/* Plug-in address */
+  CFPlugInRef	plugin;		/* Plug-in address */
```
Comment (michaelrsweet) [inline backend/usb-darwin.c:1419]: I remember this used to yield a compiler warning without the initialization.

File: backend/usb-darwin.c [hunk 2]
```diff
@@ -1429,32 +1429,34 @@ static kern_return_t load_classdriver(CFStringRef	    driverPath,
                                             _CUPS_FILE_CHECK_DIRECTORY, 1,
                                             Iterating ? NULL : _cupsFileCheckFilter, NULL);
 
-  if (result && driverPath)
-    return (load_classdriver(NULL, interface, printerDriver));
-  else if (result)
+  if (result)
```
Comment (michaelrsweet) [inline backend/usb-darwin.c:1432]: Unnecessary change, please limit macOS changes to bug fixes only (Apple isn't actually using OpenPrinting CUPS sources)

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Remove unrelated changes.

Only bug fixes for macOS-specific code, and particularly the Security.framework code is deprecated and CUPS 2.4.x prefers OpenSSL/GNU TLS over the deprecated framework...

**Comment by AtariDreams** [comment]
Merged to (https://github.com/OpenPrinting/cups/pull/655)

**Pull request closed without merging**